sycl/doc/extensions/proposed/sycl_ext_oneapi_queue_status_query.asciidoc

```diff
@@ -0,0 +1,125 @@
+= sycl_ext_intel_queue_status_querry.asciidoc
+
+:source-highlighter: coderay
+:coderay-linenums-mode: table
+
+// This section needs to be after the document title.
+:doctype: book
+:toc2:
+:toc: left
+:encoding: utf-8
+:lang: en
+:dpcpp: pass:[DPC++]
+
+// Set the default source code type in this document to C++,
+// for syntax highlighting purposes.  This is needed because
+// docbook uses c++ and html5 uses cpp.
+:language: {basebackend@docbook:c++:cpp}
+
+
+== Notice
+
+[%hardbreaks]
+Copyright (C) 2022-2022 Intel Corporation.  All rights reserved.
+
+Khronos(R) is a registered trademark and SYCL(TM) and SPIR(TM) are trademarks
+of The Khronos Group Inc.  OpenCL(TM) is a trademark of Apple Inc. used by
+permission by Khronos.
+
+
+== Contact
+
+To report problems with this extension, please open a new issue at:
+
+https://github.com/intel/llvm/issues
+
+
+== Dependencies
+
+This extension is written against the SYCL 2020 revision 4 specification.  All
+references below to the "core SYCL specification" or to section numbers in the
+SYCL specification refer to that revision.
+
+== Status
+
+This is a proposed extension specification, intended to gather community
+feedback.  Interfaces defined in this specification may not be implemented yet
+or may be in a preliminary state.  The specification itself may also change in
+incompatible ways before it is finalized.  *Shipping software products should
+not rely on APIs defined in this specification.*
+
+== Overview
+
+This extension adds `sycl::queue::size()`, `sycl::queue::empty()`, `sycl::queue::get_wait_list()` member functions.
+
+- `sycl::queue::size()` and `sycl::queue::empty` allow users to query the number of enqueued commands to a queue.
+- `sycl::queue::get_wait_list()` is the queue equivalent of `event::get_wait_list()` 
```

Comment:
It's not really necessary (or useful) to list all of the extended member functions here.  They are already described more fully below.  Instead, it's more useful to provide a justification for why this extension exists.  How do you plan to use this extension?  Can you write an overview that starts something like:

> This extension adds new query APIs to the `queue` class which allow a SYCL application to ...